Firestore/core/test/firebase/firestore/util/async_queue_test.cc

```diff
@@ -197,15 +203,20 @@ TEST_P(AsyncQueueTest, CanManuallyDrainSpecificDelayedOperationsForTesting) {
   EXPECT_EQ(steps, "1234");
 }
 
-TEST_P(AsyncQueueTest, CanScheduleOprationsWithRespectsToShutdownState) {
+TEST_P(AsyncQueueTest, CanScheduleOprationsRespectingRestrictedMode) {
   Expectation ran;
   std::string steps;
 
   queue->Enqueue([&] { steps += '1'; });
-  queue->EnqueueAndInitiateShutdown([&] { steps += '2'; });
+  queue->EnterRestrictedMode();
+  queue->EnqueueEvenWhileRestricted([&] { steps += '2'; });
   queue->Enqueue([&] { steps += '3'; });
-  queue->EnqueueEvenAfterShutdown([&] { steps += '4'; });
-  queue->EnqueueEvenAfterShutdown(ran.AsCallback());
+  queue->EnqueueEvenWhileRestricted([&] { steps += '4'; });
+  queue->EnqueueEvenWhileRestricted(ran.AsCallback());
+
+  queue->Stop();
+  queue->Enqueue([&] { steps += '5'; });
+  queue->EnqueueEvenWhileRestricted([&] { steps += '6'; });
```

Comment:
No action required: technically, I think this test could pass even if the queue actually executed both of the operations enqueued after shutdown (because there is no synchronization to wait for them). On the other hand, I don't think it's possible to add that synchronization, so I guess there's not much we can do.